lizmap/modules/view/controllers/media.classic.php

```diff
@@ -287,11 +287,34 @@ public function getMedia()
             $rep->fileName = '';
             $rep->content = $content;
         }
+        // For HEAD request, we need to set the content length
+        $fileSize = filesize($finalPath);
+        $rep->addHttpHeader('Content-Length', $fileSize);
 
-        $rep->setExpires('+1 days');
+        // Set Allow header to GET and HEAD for all files
+        $rep->addHttpHeader('Allow', 'GET, HEAD');
 
         if ($isWebDavResource) {
+            $rep->setExpires('+1 days');
             $rep->deleteFileAfterSending = true;
+        } else {
+            // Etag header and cache control
+            $etag = '';
+            if ($this->canBeCached()) {
+                $etag = 'media';
+                $etag .= '-'.$lrep->getKey().'~'.$lproj->getKey();
+                $etag .= '-'.$path;
+                $etag .= '-'.filemtime($finalPath);
+                $etag = sha1($etag);
+            }
+
+            if ($etag !== '' && $rep->isValidCache(null, $etag)) {
+                return $rep;
+            }
+
+            if ($etag !== '') {
+                $this->setEtagCacheHeaders($rep, $etag);
+            }
```

Comment:
I think we could simplify code like this:

```suggestion
        } else  if ($this->canBeCached()) {
            // Etag header and cache control
            $etag = 'media';
            $etag .= '-'.$lrep->getKey().'~'.$lproj->getKey();
            $etag .= '-'.$path;
            $etag .= '-'.filemtime($finalPath);
            $etag = sha1($etag);

            if ($rep->isValidCache(null, $etag)) {
                return $rep;
            }

            $this->setEtagCacheHeaders($rep, $etag);
```